src/Contexts/UserContext.jsx

```diff
@@ -95,13 +95,102 @@ const UserProvider = ({ children }) => {
           const {
             data: { history },
           } = await axios.delete(`/api/user/history/${video._id}`, { headers: { authorization: authState.token } });
-          console.log(history);
         } catch (err) {
           console.log(err);
         }
       })();
     }
   };
+  const RemoveAllVideoFromHistory = () => {
+    if (authState.isUserLoggedIn) {
+      dispatch({ type: "Remove_All_Video_From_History" });
+      (async () => {
+        try {
+          const {
+            data: { history },
+          } = await axios.delete(`/api/user/history/all`, { headers: { authorization: authState.token } });
+        } catch (err) {
+          console.log(err);
+        }
+      })();
+    }
+  };
+  const CreatePlaylist = (PlayListTitle) => {
+    if (authState.isUserLoggedIn) {
+      (async () => {
+        try {
+          const {
+            data: { playlists },
+          } = await axios.post(
+            "/api/user/playlists",
+            {
+              playlist: { title: PlayListTitle, description: "PlayList" },
+            },
+            { headers: { authorization: authState.token } }
+          );
+          dispatch({ type: "Create_PlayList", payload: playlists });
+        } catch (error) {
+          console.log(error);
+        }
+      })();
+    }
+  };
+  const DeletePlaylist = (playList) => {
+    if (authState.isUserLoggedIn) {
+      dispatch({ type: "Delete_PlayList", payload: playList });
+      (async () => {
+        try {
+          const {
+            data: { playlists },
+          } = await axios.delete(`/api/user/playlists/${playList._id}`, {
```

Comment:
You can use separate(services) file for calling API

Reply:
OK, I will create.